Next GitHub docs (#986)
tinacms/tinacms: packages/next-tinacms-github/README.md

)
```

> [See below](#github-oauth-app) for creating a Github Oauth App provides these **Client ID** & **Client Secret** variables.

Reviewer: ```suggestion
[See below](#github-oauth-app) for instructions on creating a Github OAuth App to generate these **Client ID** & **Client Secret** variables.
```

import { createPreviewFn } from 'next-tinacms-github'

> const fork_full_name = 'https://github.com/username/reponame'

Reviewer: Maybe `forkUrl`? 

Also camelCase instead of snake_case

      previewData.accessToken)
    }
    catch (e) {
      // If there is an error initially loading the content from Github, we want to display an actionable error
      // to the user. They may need to re-authenticate or create a new fork.
      if (e instanceof GithubError) {
        error = { ...e } //workaround since we cant return error as JSON
      } else {
        throw e
      }
    }
  }
  else {
    // Get your production content here
    // when you are not in edit-mode
     file = await readLocalMarkdownFile(filePath)
  }

  return {
    props: {
      sourceProviderConnection,
      editMode: !!preview,
      file,
      error
    },
  }
}
```

### Github Oauth App:

In GitHub, within your account Settings, click [Oauth Apps](https://github.com/settings/developers) under Developer Settings.

click "New Oauth App".

For the **Authorization callback URL**, enter the url for the "authorizing" page that you created above (e.g https://your-url/github/authorizing). Fill out the other fields with your custom values.
_Note: If you are testing your app locally, you may need a separate development Github app (with a localhost redirect), and a production Github app._ 

> The generated **Client ID** & **Client Secret** will be consumed by the `createCreateAccessTokenFn` [defined above](#createcreateaccesstokenfn).

Reviewer: Maybe add a bit to this section about how to use those variables. Specifically

1. How to add them to your environment variables 
2. How to register them in the `next.config.js`

export default createProxy(GITHUB_ACCESS_TOKEN_COOKIE_KEY)
```

### `createPreviewFn`

Helper for creating a preview function which will set the Nextjs [preview data](https://nextjs.org/docs/advanced-features/preview-mode) from your cookie data.

```
// pages/api/preview.ts

import { createPreviewFn } from 'next-tinacms-github'

const forkCookieKey = 'fork_full_name'
const headBranchCookieKey =  'head_branch'
const githubAccessTokenCookieKey = 'github_access_token'

export default createPreviewFn(
  forkFullName,
  headBranch,
  githubAccessToken
)
```

### Loading content from Github

This preview data set from calling your [preview function](#createpreviewfn) will be accesible through `getStaticProps` throughout your app.

```ts
//Blog template [slug.ts]

import {
  getMarkdownFile as getGithubMarkdownFile,
} from 'next-tinacms-github'

// ...

export const getStaticProps: GetStaticProps = async function({
  preview,
  previewData,
  ...ctx
}) {
  const { slug } = ctx.params

  let file = {}
  const filePath = `content/blog/${slug}.md`

  const sourceProviderConnection = {
    forkFullName: previewData.fork_full_name,
    headBranch: previewData.head_branch || 'master', 
  }

>   if(preview)
>   {

Reviewer: Egyptian brackets
```suggestion
  if(preview) {
```

    file = await getGithubMarkdownFile(filePath,
      sourceProviderConnection, 
      previewData.accessToken)
  }

>   else 
>   {

Reviewer: Egyptian brackets
```suggestion
  else {
```

    // Get your production content here
    // when you are not in edit-mode
     file = await readLocalMarkdownFile(filePath)
  }

  return {
    props: {
      sourceProviderConnection
      editMode: !!preview,
      file,

>     },

Reviewer: Is the `previewError` no longer needed?
